src/BitStringAddresses/bosefs.jl

```diff
@@ -1,112 +1,113 @@
+
+* `BoseFS{N,M,S}(bs::S)`: Unsafe constructor. Does not check whether the number of
   particles in `bs` is equal to `N`.
 
-* `BoseFS(::BitString)`: Automatically determine `N` and `M`. This constructor is not type
-  stable!
+* [`@fs_str`](@ref): addresses are sometimes printed in a compact manner. This
+  representation can also be used as a constructor. See the last example below.
 
-* `BoseFS{[N,M,S]}(onr)`: Create `BoseFS{N,M}` from [`onr`](@ref) representation. This is
-  efficient as long as at least `N` is provided.
+# Examples
 
-See also: [`SingleComponentFockAddress`](@ref), [`FermiFS`](@ref), [`BitString`](@ref).
-"""
-struct BoseFS{N,M,S<:BitString} <: SingleComponentFockAddress{N,M}
-    bs::S
-end
+```jldoctest
+julia> BoseFS{6,5}((0, 1, 2, 3, 0))
+BoseFS{6,5}((0, 1, 2, 3, 0))
 
-function BoseFS{N,M}(bs::BitString{B}) where {N,M,B}
-    # Check for consistency between parameter, but NOT for the correct number of bits.
-    N + M - 1 == B || throw(ArgumentError("type parameter mismatch"))
-    return BoseFS{N,M,typeof(bs)}(bs)
-end
+julia> BoseFS([abs(i - 3) ≤ 1 ? i - 1 : 0 for i in 1:5])
+BoseFS{6,5}((0, 1, 2, 3, 0))
 
-function BoseFS(bs::BitString{B}) where B
-    N = count_ones(bs)
-    M = B - N + 1
-    return BoseFS{N,M}(bs)
-end
+julia> BoseFS(5, 2 => 1, 3 => 2, 4 => 3)
+BoseFS{6,5}((0, 1, 2, 3, 0))
 
-@inline function BoseFS{N,M,S}(
-    onr::Union{SVector{M},NTuple{M}}
-) where {N,M,S<:BitString{<:Any,1}}
-    @boundscheck sum(onr) == N || error("invalid ONR")
-    T = chunk_type(S)
-    result = zero(T)
-    for i in M:-1:1
-        curr_occnum = T(onr[i])
-        result <<= curr_occnum + T(1)
-        result |= one(T) << curr_occnum - T(1)
-    end
-    return BoseFS{N,M,S}(S(SVector(result)))
-end
+julia> BoseFS{6,5}(i => i - 1 for i in 2:4)
+BoseFS{6,5}((0, 1, 2, 3, 0))
 
-@inline function BoseFS{N,M,S}(onr::Union{SVector{M},NTuple{M}}) where {N,M,S<:BitString}
-    @boundscheck sum(onr) == N || error("invalid ONR")
-    K = num_chunks(S)
-    result = zeros(MVector{K,UInt64})
-    offset = 0
-    bits_left = chunk_bits(S, K)
-    i = 1
-    j = K
-    while true
-        # Write number to result
-        curr_occnum = onr[i]
-        while curr_occnum > 0
-            x = min(curr_occnum, bits_left)
-            mask = (one(UInt64) << x - 1) << offset
-            @inbounds result[j] |= mask
-            bits_left -= x
-            offset += x
-            curr_occnum -= x
-
-            if bits_left == 0
-                j -= 1
-                offset = 0
-                bits_left = chunk_bits(S, j)
-            end
-        end
-        offset += 1
-        bits_left -= 1
+julia> fs"|0 1 2 3 0⟩"
+BoseFS{6,5}((0, 1, 2, 3, 0))
 
-        if bits_left == 0
-            j -= 1
-            offset = 0
-            bits_left = chunk_bits(S, j)
-        end
-        i += 1
-        i > M && break
-    end
-    return BoseFS{N,M}(S(SVector(result)))
+julia> fs"|b 5: 2 3 3 4 4 4⟩"
+BoseFS{6,5}((0, 1, 2, 3, 0))
```

Comment:
I know we didn't settle on what the format should be, but the prefix for this constructor should be different than "fs", to avoid confusion with the `fs"|0 1 2 3 0⟩"` which is also a valid constructor

Reply:
It's nicer if it's `fs` because it allows you to show mixed addresses as e.g. `fs"|1 1 1 1 1 1⟩ ⊗ |b 6: 2⟩"`.